experimental/packages/opentelemetry-instrumentation-http/src/utils.ts

```diff
@@ -248,14 +311,27 @@ export const getRequestInfo = (
   origin: string;
   pathname: string;
   method: string;
+  invalidUrl: boolean;
   optionsParsed: RequestOptions;
 } => {
-  let pathname = '/';
-  let origin = '';
+  let pathname: string;
+  let origin: string;
   let optionsParsed: RequestOptions;
+  let invalidUrl = false;
   if (typeof options === 'string') {
-    optionsParsed = url.parse(options);
-    pathname = (optionsParsed as url.UrlWithStringQuery).pathname || '/';
+    try {
+      const convertedOptions = stringUrlToHttpOptions(options);
+      optionsParsed = convertedOptions;
+      pathname = convertedOptions.pathname || '/';
+    } catch (e) {
+      invalidUrl = true;
```

Comment:
This will be executed when TypeError is returned by URL() for invalid inputs right? we should log internally about it

Reply:
Yep, good idea. 👍 
I'm now also passing in the component logger from the instrumentation: https://github.com/open-telemetry/opentelemetry-js/pull/5091/commits/4c5825ca6d2a9da90a9defcb38e2e3117280524e